src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -27,7 +30,12 @@ namespace
     std::mutex g_lib_lock;
     std::weak_ptr<coreclr_t> g_lib_coreclr;
 
-    int create_coreclr(const hostpolicy_context_t &context, host_mode_t mode, std::unique_ptr<coreclr_t> &coreclr)
+    std::mutex g_context_lock;
```

Comment:
I'm wondering why we need the two set of locks and globals. Why isn't the `coreclr_t` part of the `hostpolicy_context_t`? Then we would only need one synchronization mechanism... having these two... it just smells like asking for trouble with deadlocks.

Reply:
Note that all this synchronization is new in 3.0, so there are limited backward compat concerns

Reply:
Yay, added `coreclr_t` to `hostpolicy_context_t`.